### ISSUE TITLE
Request - Adding PGP to Roundcube

### ISSUE THREAD

**Issue by lbouriez** [Purelymail/purelymail-issues#105]
As mentioned on Discord, could be nice to allow us to use PGP key directly into the webmail.
Vitor mentionned this plugin: https://github.com/roundcube/roundcubemail/tree/master/plugins/enigma

**Comment by ScottPeterJohnson** [comment]
I'll take a look at the plugin quality, but I'm unsure whether this is a good idea- since adding the PGP into webmail _invalidates the point of PGP keys._

**Comment by snevas** [comment]
I think you both should look into [mailvelope](https://mailvelope.com/en). That's a browser plugin that handles pgp locally (optionally also with GnuPG and yubikey support). There are multiple roundcube plugins available to handle the client side api calls from roundcube to the extension.
Most mayor mailhosters already built support for mailvelope in their products.

**Comment by impurify** [comment]
> ...adding the PGP into webmail _invalidates the point of PGP keys._

Good call.  The point of PGP keys includes minimizing trust in the server!  And I’d suggest considering whether it could lose more users than it could gain.  I distrust mail services that solicit naïve users to expose their PGP private keys to the mail or webmail server, and I’m not the only one.

It also paints a target on the server and its administrator, for anyone who may want to obtain access to those keys.  Do you want to take responsibility for promising users that you will *never* knowingly or unknowingly expose their keys for any reason?

@lbouriez, please consider using PGP the only secure way it can be used:  Locally, on your own machine.  Placing your PGP private keys on the server gives a false sense of security both to you, *and to your unknowing correspondents to whom you are falsely pretending E2EE secure communications*.

Beyond @snevas’s suggestion of Mailvelope, there are [free software tools for using GnuPG](https://gnupg.org/software/swlist.html) on a variety of platforms, including [Microsoft Windows](https://gpg4win.org/) and [Apple Mac](https://gpgtools.org/); it should go without saying that there are plenty of options on Linux, the BSDs, and other free and open source platforms.  HTH.
